### ISSUE TITLE
Matrix predictors (was No variable named [X] in the reference grid)

### ISSUE THREAD

**Issue by mattansb** [rvlenth/emmeans#66]
I found this funny little annoyance when centering / scaling variables with the `base::scale` function that seems to affect `emmeans` (but not `emtrends` for some reason):

Setup
````
> library(emmeans)
> set.seed(143)
> 
> Y <- rnorm(100)
> G <- rep(letters[1:2], each = 50)
> X <- rnorm(100)
````

When using `scale`:
````
> Xc <- scale(X)
> mo1 <- lm(Y ~ G*Xc)
> emtrends(mo1, ~G, 'Xc')    # works
 G    Xc.trend        SE df   lower.CL  upper.CL
 a  0.02633894 0.1510146 96 -0.2734226 0.3261005
 b -0.17649041 0.1332786 96 -0.4410464 0.0880656

Confidence level used: 0.95 
> emmeans(mo1, ~Xc|G) # errors
Error in emmeans(mo1, ~Xc | G) : 
  No variable named Xc in the reference grid
````

The solution - using `scale` with `as.numeric`:
````
> Xc2 <- as.numeric(scale(X))
> mo2 <- lm(Y ~ G*Xc2)
> emtrends(mo2, ~G, 'Xc2')     # works
 G   Xc2.trend        SE df   lower.CL  upper.CL
 a  0.02633894 0.1510146 96 -0.2734226 0.3261005
 b -0.17649041 0.1332786 96 -0.4410464 0.0880656

Confidence level used: 0.95 
> emmeans(mo2, ~Xc2|G)    # works
G = a:
           Xc2    emmean        SE df    lower.CL  upper.CL
 -1.482647e-17 0.2320536 0.1406112 96 -0.04705736 0.5111646

G = b:
           Xc2    emmean        SE df    lower.CL  upper.CL
 -1.482647e-17 0.1156334 0.1406095 96 -0.16347438 0.3947411

Confidence level used: 0.95 
````

I suspect this is because the numeric vector now has attributes? But I was unable to replicate this error when just adding attributes to vectors, so it might not be that. Anyway, just posting this here for any other poor saps who encounter it.

**Comment by rvlenth** [comment]
However, here's something interesting. If you specify values for `Xc` in `at`, it works fine:
```
> emmeans(mo1, ~Xc|G, at = list(Xc = -1:1))
G = a:
 Xc  emmean    SE df lower.CL upper.CL
 -1  0.2057 0.205 96  -0.2018    0.613
  0  0.2321 0.141 96  -0.0471    0.511
  1  0.2584 0.207 96  -0.1533    0.670

G = b:
 Xc  emmean    SE df lower.CL upper.CL
 -1  0.2921 0.195 96  -0.0942    0.678
  0  0.1156 0.141 96  -0.1635    0.395
  1 -0.0609 0.193 96  -0.4437    0.322

Confidence level used: 0.95
```
I'm not saying this is desirable either, because it overrides the fact that Xc is a matrix. Thus, that in itself is another error...

**Comment by rvlenth** [comment]
I have just pushed a revision that sort of addresses this issue. A one-column matrix predictor is now treated just like a regular covariate. But a multi-column matrix still has a kind of second-class status: not really part of the reference grid except for obtaining predictions. I have fixed things enough so that errors don't occur in summarizing a ref grid with matrix predictors, and added some documentation to make it clearer what happens to matrices.

At some future time I may add first-class support for matrices; but don't hold your breath.
